### PR TITLE
Materials subsystem, material defines instead of strings, structure materials.

### PR DESCRIPTION
**Pull request by NonQueueingMatt**
This is all backend changes for now, but it sets the stage for further material integration into most items and structures.

Also cleans up materials and such, more to do in another PR. Jesus christ what a mess.

### CODE REVIEW

**Comment by NonQueueingMatt** [comment]
!wip

**Comment by NonQueueingMatt** [comment]
!review

File: code/game/objects/items/devices/radio/radio.dm [hunk 1]
```diff
@@ -44,7 +44,7 @@ var/global/list/default_medbay_channels = list(
 	throw_speed = 2
 	throw_range = 9
 	w_class = 2
-	matter = list("glass" = 25,DEFAULT_WALL_MATERIAL = 75)
+	matter = list(MATERIAL_GLASS = 25,DEFAULT_WALL_MATERIAL = 75)
```
Comment (Geevies) [inline code/game/objects/items/devices/radio/radio.dm]:
```suggestion
	matter = list(DEFAULT_WALL_MATERIAL = 75, MATERIAL_GLASS = 25)
```

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 2]
```diff
@@ -130,7 +130,7 @@
 /obj/item/borg/upgrade/combat
 	name = "combat cyborg module"
 	desc = "Unlocks the combat cyborg module"
-//	construction_cost = list(DEFAULT_WALL_MATERIAL=10000,"glass"=15000,"gold"= 5000,"diamond" = 1000)
+//	construction_cost = list(DEFAULT_WALL_MATERIAL=10000, MATERIAL_GLASS =15000, MATERIAL_GOLD = 5000, MATERIAL_DIAMOND = 1000)
```
Comment (Geevies) [inline code/game/objects/items/robot/robot_upgrades.dm]:
```suggestion
```

File: code/modules/assembly/igniter.dm [hunk 3]
```diff
@@ -3,7 +3,7 @@
 	desc = "A small electronic device able to ignite combustable substances."
 	icon_state = "igniter"
 	origin_tech = list(TECH_MAGNET = 1)
-	matter = list(DEFAULT_WALL_MATERIAL = 500, "glass" = 50, "waste" = 10)
+	matter = list(DEFAULT_WALL_MATERIAL = 500, MATERIAL_GLASS = 50, "waste" = 10)
```
Comment (Geevies) [inline code/modules/assembly/igniter.dm]: waste?

File: code/modules/clothing/spacesuits/rig/modules/computer.dm [hunk 4]
```diff
@@ -36,7 +36,7 @@
 	activates_on_touch = 1
 	confined_use = 1
 
-	construction_cost = list("glass" = 7500, DEFAULT_WALL_MATERIAL = 5000)
+	construction_cost = list(MATERIAL_GLASS = 7500, DEFAULT_WALL_MATERIAL = 5000)
```
Comment (Geevies) [inline code/modules/clothing/spacesuits/rig/modules/computer.dm]:
```suggestion
	construction_cost = list(DEFAULT_WALL_MATERIAL = 5000, MATERIAL_GLASS = 7500)
```

File: code/modules/clothing/spacesuits/rig/modules/utility.dm [hunk 5]
```diff
@@ -64,7 +64,7 @@
 	suit_overlay_active = "mounted-drill"
 	suit_overlay_inactive = "mounted-drill"
 	use_power_cost = 0.1
-	construction_cost = list("glass"=2250,DEFAULT_WALL_MATERIAL=55000,"silver"=5250,"diamond"=3750)
+	construction_cost = list(MATERIAL_GLASS = 2250, DEFAULT_WALL_MATERIAL = 55000, MATERIAL_SILVER = 5250, MATERIAL_DIAMOND = 3750)
```
Comment (Geevies) [inline code/modules/clothing/spacesuits/rig/modules/utility.dm]:
```suggestion
	construction_cost = list(DEFAULT_WALL_MATERIAL = 55000, MATERIAL_GLASS = 2250, MATERIAL_SILVER = 5250, MATERIAL_DIAMOND = 3750)
```

File: code/modules/clothing/spacesuits/rig/modules/utility.dm [hunk 6]
```diff
@@ -424,15 +424,14 @@
...
-
 	name = "hardsuit maneuvering jets"
 	desc = "A compact gas thruster system for a hardsuit."
 	icon_state = "thrusters"
 	usable = 1
 	toggleable = 1
 	selectable = 0
 	disruptive = 0
-	construction_cost = list("glass"= 4250,DEFAULT_WALL_MATERIAL=15000,"silver"=4250,"uranium"=5250)
+	construction_cost = list(MATERIAL_GLASS = 4250, DEFAULT_WALL_MATERIAL = 15000, MATERIAL_SILVER = 4250, MATERIAL_URANIUM = 5250)
```
Comment (Geevies) [inline code/modules/clothing/spacesuits/rig/modules/utility.dm]:
```suggestion
	construction_cost = list(DEFAULT_WALL_MATERIAL = 15000, MATERIAL_GLASS = 4250, MATERIAL_SILVER = 4250, MATERIAL_URANIUM = 5250)
```

File: code/modules/clothing/spacesuits/rig/modules/vision.dm [hunk 7]
```diff
@@ -104,7 +104,7 @@
 
 	usable = 0
 
-	construction_cost = list("glass"=5000,DEFAULT_WALL_MATERIAL=1500)
+	construction_cost = list(MATERIAL_GLASS =5000,DEFAULT_WALL_MATERIAL=1500)
```
Comment (Geevies) [inline code/modules/clothing/spacesuits/rig/modules/vision.dm]:
```suggestion
	construction_cost = list(DEFAULT_WALL_MATERIAL = 1500, MATERIAL_GLASS = 5000)
```

File: code/modules/clothing/spacesuits/rig/modules/vision.dm [hunk 8]
```diff
@@ -136,7 +136,7 @@
 
 	usable = 0
 
-	construction_cost = list("glass"=5000,DEFAULT_WALL_MATERIAL=1500,"uranium"=5000)
+	construction_cost = list(MATERIAL_GLASS = 5000,DEFAULT_WALL_MATERIAL = 1500, MATERIAL_URANIUM = 5000)
```
Comment (Geevies) [inline code/modules/clothing/spacesuits/rig/modules/vision.dm]:
```suggestion
	construction_cost = list(DEFAULT_WALL_MATERIAL = 1500, MATERIAL_GLASS = 5000, MATERIAL_URANIUM = 5000)
```

File: code/modules/clothing/spacesuits/rig/modules/vision.dm [hunk 9]
```diff
@@ -154,7 +154,7 @@
 
 	usable = 0
 
-	construction_cost = list("glass"=5000,DEFAULT_WALL_MATERIAL =1500)
+	construction_cost = list(MATERIAL_GLASS =5000,DEFAULT_WALL_MATERIAL =1500)
```
Comment (Geevies) [inline code/modules/clothing/spacesuits/rig/modules/vision.dm]:
```suggestion
	construction_cost = list(DEFAULT_WALL_MATERIAL = 1500, MATERIAL_GLASS = 5000)
```

File: code/modules/clothing/spacesuits/rig/modules/vision.dm [hunk 10]
```diff
@@ -172,7 +172,7 @@
 
 	usable = 0
 
-	construction_cost = list("glass"=5000,DEFAULT_WALL_MATERIAL=1500)
+	construction_cost = list(MATERIAL_GLASS =5000,DEFAULT_WALL_MATERIAL=1500)
```
Comment (Geevies) [inline code/modules/clothing/spacesuits/rig/modules/vision.dm]:
```suggestion
	construction_cost = list(DEFAULT_WALL_MATERIAL = 1500, MATERIAL_GLASS = 5000)
```

File: code/modules/integrated_electronics/core/tools.dm [hunk 11]
```diff
@@ -250,7 +250,7 @@
 		"orange" = COLOR_ASSEMBLY_ORANGE,
 		"beige" = COLOR_ASSEMBLY_BEIGE,
 		"brown" = COLOR_ASSEMBLY_BROWN,
-		"gold" = COLOR_ASSEMBLY_GOLD,
+		MATERIAL_GOLD = COLOR_ASSEMBLY_GOLD,
```
Comment (Geevies) [inline code/modules/integrated_electronics/core/tools.dm]: not a material

File: code/modules/mining/drilling/drill.dm [hunk 12]
```diff
@@ -18,12 +18,12 @@
 
 	var/ore_types = list(
 		"iron" = /obj/item/ore/iron,
-		"uranium" = /obj/item/ore/uranium,
-		"gold" = /obj/item/ore/gold,
-		"silver" = /obj/item/ore/silver,
-		"diamond" = /obj/item/ore/diamond,
-		"phoron" = /obj/item/ore/phoron,
-		"osmium" = /obj/item/ore/osmium,
+		MATERIAL_URANIUM = /obj/item/ore/uranium,
```
Comment (Geevies) [inline code/modules/mining/drilling/drill.dm]: im not 100% sure about changing these

File: code/modules/research/designs/mechfab/hardsuit/modules.dm [hunk 13]
```diff
@@ -8,125 +8,125 @@
 	desc = "An integrated intelligence system module suitable for most hardsuits."
 	id = "iis_module"
 	req_tech = list(TECH_DATA = 4, TECH_MATERIAL = 3)
-	materials = list("glass" = 7500, DEFAULT_WALL_MATERIAL = 5000)
+	materials = list(MATERIAL_GLASS = 7500, DEFAULT_WALL_MATERIAL = 5000)
```
Comment (Geevies) [inline code/modules/research/designs/mechfab/hardsuit/modules.dm]:
```suggestion
	materials = list(DEFAULT_WALL_MATERIAL = 5000, MATERIAL_GLASS = 7500)
```

File: code/modules/research/designs/mechfab/hardsuit/modules.dm [hunk 14]
```diff
@@ -8,125 +8,125 @@
...
+	materials = list(DEFAULT_WALL_MATERIAL = 10000, MATERIAL_GOLD = 2000, MATERIAL_SILVER = 3000, MATERIAL_GLASS = 2000)
 	build_path = /obj/item/rig_module/power_sink
 
 /datum/design/hardsuitmodules/meson_module
 	name = "hardsuit meson scanner"
 	desc = "A layered, translucent visor system for a hardsuit."
 	id = "meson_module"
 	req_tech = list(TECH_MAGNET = 4, TECH_MATERIAL = 2, TECH_ENGINEERING = 3)
-	materials = list("glass"= 5000, DEFAULT_WALL_MATERIAL= 1500)
+	materials = list(MATERIAL_GLASS = 5000, DEFAULT_WALL_MATERIAL= 1500)
```
Comment (Geevies) [inline code/modules/research/designs/mechfab/hardsuit/modules.dm]:
```suggestion
	materials = list(DEFAULT_WALL_MATERIAL = 1500, MATERIAL_GLASS = 5000)
```

File: code/modules/research/designs/mechfab/hardsuit/modules.dm [hunk 15]
```diff
@@ -8,125 +8,125 @@
...
 	build_path = /obj/item/rig_module/vision/meson
 
 /datum/design/hardsuitmodules/sechud_module
 	name = "hardsuit security hud"
 	desc = "A simple tactical information system for a hardsuit."
 	id = "sechud_module"
 	req_tech = list(TECH_BIO = 3, TECH_MATERIAL = 2, TECH_MAGNET = 3)
-	materials = list("glass" = 5000, DEFAULT_WALL_MATERIAL =1500)
+	materials = list(MATERIAL_GLASS = 5000, DEFAULT_WALL_MATERIAL =1500)
```
Comment (Geevies) [inline code/modules/research/designs/mechfab/hardsuit/modules.dm]:
```suggestion
	materials = list(DEFAULT_WALL_MATERIAL = 1500, MATERIAL_GLASS = 5000)
```

File: code/modules/research/designs/mechfab/hardsuit/modules.dm [hunk 16]
```diff
@@ -8,125 +8,125 @@
...
 	build_path = /obj/item/rig_module/vision/sechud
 
 /datum/design/hardsuitmodules/medhud_module
 	name = "hardsuit medical hud"
 	desc = "A simple medical status indicator for a hardsuit."
 	id = "medhu_module"
 	req_tech = list(TECH_BIO = 3, TECH_MATERIAL = 2, TECH_MAGNET = 3)
-	materials = list("glass"= 5000, DEFAULT_WALL_MATERIAL =1500)
+	materials = list(MATERIAL_GLASS = 5000, DEFAULT_WALL_MATERIAL =1500)
```
Comment (Geevies) [inline code/modules/research/designs/mechfab/hardsuit/modules.dm]:
```suggestion
	materials = list(DEFAULT_WALL_MATERIAL = 1500, MATERIAL_GLASS = 5000)
```

File: code/modules/research/designs/mechfab/hardsuit/modules.dm [hunk 17]
```diff
@@ -8,125 +8,125 @@
...
 	build_path = /obj/item/rig_module/vision/medhud
 
 /datum/design/hardsuitmodules/nvg_module
 	name = "hardsuit night vision interface"
 	desc = "A multi input night vision system for a hardsuit."
 	id = "nvg_module"
 	req_tech = list(TECH_BIO = 4, TECH_MATERIAL = 3, TECH_MAGNET = 4)
-	materials = list("glass" = 5000, DEFAULT_WALL_MATERIAL = 1500, "uranium" = 5000)
+	materials = list(MATERIAL_GLASS = 5000, DEFAULT_WALL_MATERIAL = 1500, MATERIAL_URANIUM = 5000)
```
Comment (Geevies) [inline code/modules/research/designs/mechfab/hardsuit/modules.dm]:
```suggestion
	materials = list(DEFAULT_WALL_MATERIAL = 1500, MATERIAL_GLASS = 5000, MATERIAL_URANIUM = 5000)
```

File: code/modules/research/designs/mechfab/hardsuit/modules.dm [hunk 18]
```diff
@@ -8,125 +8,125 @@
...
 	build_path = /obj/item/rig_module/vision/nvg
 
 /datum/design/hardsuitmodules/healthscanner_module
 	name = "hardsuit health scanner"
 	desc = "A hardsuit-mounted health scanner."
 	id = "healthscanner_module"
 	req_tech = list(TECH_BIO = 3, TECH_MATERIAL = 3, TECH_MAGNET = 2)
-	materials = list("glass" = 5250, DEFAULT_WALL_MATERIAL = 2500)
+	materials = list(MATERIAL_GLASS = 5250, DEFAULT_WALL_MATERIAL = 2500)
```
Comment (Geevies) [inline code/modules/research/designs/mechfab/hardsuit/modules.dm]:
```suggestion
	materials = list(DEFAULT_WALL_MATERIAL = 2500, MATERIAL_GLASS = 5250)
```

File: code/modules/research/designs/mechfab/hardsuit/modules.dm [hunk 19]
```diff
@@ -8,125 +8,125 @@
...
 	build_path = /obj/item/rig_module/device/healthscanner
 
 /datum/design/hardsuitmodules/chem_module
 	name = "mounted chemical injector"
 	desc = "A complex web of tubing and a large needle suitable for hardsuit use."
 	id = "chem_module"
 	req_tech = list(TECH_BIO = 5, TECH_MATERIAL = 4, TECH_DATA = 3, TECH_PHORON = 2)
-	materials = list("glass" = 9250, DEFAULT_WALL_MATERIAL = 10000, "gold" = 2500, "silver" = 4250, "phoron" = 5500)
+	materials = list(MATERIAL_GLASS = 9250, DEFAULT_WALL_MATERIAL = 10000, MATERIAL_GOLD = 2500, MATERIAL_SILVER = 4250, MATERIAL_PHORON = 5500)
```
Comment (Geevies) [inline code/modules/research/designs/mechfab/hardsuit/modules.dm]:
```suggestion
	materials = list(DEFAULT_WALL_MATERIAL = 10000, MATERIAL_GLASS = 9250, MATERIAL_GOLD = 2500, MATERIAL_SILVER = 4250, MATERIAL_PHORON = 5500)
```

File: code/modules/research/designs/mechfab/hardsuit/modules.dm [hunk 20]
```diff
@@ -8,125 +8,125 @@
...
 	build_path = /obj/item/rig_module/chem_dispenser/injector
 
 /datum/design/hardsuitmodules/plasmacutter_module
 	name = "hardsuit plasma cutter"
 	desc = "A self-sustaining plasma arc capable of cutting through walls."
 	id = "plasmacutter_module"
 	req_tech = list(TECH_ENGINEERING = 4, TECH_MATERIAL = 3, TECH_PHORON = 4)
-	materials = list("glass" = 5250, DEFAULT_WALL_MATERIAL = 30000, "silver" = 5250, "phoron" = 7250)
+	materials = list(MATERIAL_GLASS = 5250, DEFAULT_WALL_MATERIAL = 30000, MATERIAL_SILVER = 5250, MATERIAL_PHORON = 7250)
```
Comment (Geevies) [inline code/modules/research/designs/mechfab/hardsuit/modules.dm]:
```suggestion
	materials = list(DEFAULT_WALL_MATERIAL = 30000, MATERIAL_GLASS = 5250, MATERIAL_SILVER = 5250, MATERIAL_PHORON = 7250)
```

File: code/modules/research/designs/mechfab/hardsuit/modules.dm [hunk 21]
```diff
@@ -8,125 +8,125 @@
...
 	build_path = /obj/item/rig_module/mounted/plasmacutter
 
 /datum/design/hardsuitmodules/jet_module
 	name = "hardsuit maneuvering jets"
 	desc = "A compact gas thruster system for a hardsuit."
 	id = "jet_module"
 	req_tech = list(TECH_ENGINEERING = 4, TECH_MATERIAL = 3, TECH_POWER = 2)
-	materials = list("glass" = 4250, DEFAULT_WALL_MATERIAL = 15000, "silver" = 4250,"uranium" = 5250)
+	materials = list(MATERIAL_GLASS = 4250, DEFAULT_WALL_MATERIAL = 15000, MATERIAL_SILVER = 4250, MATERIAL_URANIUM = 5250)
```
Comment (Geevies) [inline code/modules/research/designs/mechfab/hardsuit/modules.dm]:
```suggestion
	materials = list(DEFAULT_WALL_MATERIAL = 15000, MATERIAL_GLASS = 4250, MATERIAL_SILVER = 4250, MATERIAL_URANIUM = 5250)
```

File: code/modules/research/designs/mechfab/hardsuit/modules.dm [hunk 22]
```diff
@@ -8,125 +8,125 @@
...
 	build_path = /obj/item/rig_module/maneuvering_jets
 
 /datum/design/hardsuitmodules/drill_module
 	name = "hardsuit drill mount"
 	desc = "A very heavy diamond-tipped drill."
 	id = "drill_module"
 	req_tech = list(TECH_ENGINEERING = 5, TECH_MATERIAL = 5, TECH_POWER = 4, TECH_MAGNET = 4)
-	materials = list("glass" = 2250, DEFAULT_WALL_MATERIAL = 55000, "silver" = 5250, "diamond" = 3750)
+	materials = list(MATERIAL_GLASS = 2250, DEFAULT_WALL_MATERIAL = 55000, MATERIAL_SILVER = 5250, MATERIAL_DIAMOND = 3750)
```
Comment (Geevies) [inline code/modules/research/designs/mechfab/hardsuit/modules.dm]:
```suggestion
	materials = list(DEFAULT_WALL_MATERIAL = 55000, MATERIAL_GLASS = 2250, MATERIAL_SILVER = 5250, MATERIAL_DIAMOND = 3750)
```

File: code/modules/research/designs/mechfab/hardsuit/modules.dm [hunk 23]
```diff
@@ -8,125 +8,125 @@
...
 	build_path = /obj/item/rig_module/device/drill
 
 /datum/design/hardsuitmodules/rfd_c_module
 	name = "RFD-C mount"
 	desc = "A cell-powered Rapid-Fabrication-Device C-Class for a hardsuit."
 	id = "rcd_module"
 	req_tech = list(TECH_ENGINEERING = 6, TECH_MATERIAL = 5, TECH_POWER = 5, TECH_BLUESPACE = 4)
-	materials = list(DEFAULT_WALL_MATERIAL= 30000, "phoron" = 12500, "silver" = 10000, "gold" = 10000)
+	materials = list(DEFAULT_WALL_MATERIAL= 30000, MATERIAL_PHORON = 12500, MATERIAL_SILVER = 10000, MATERIAL_GOLD = 10000)
```
Comment (Geevies) [inline code/modules/research/designs/mechfab/hardsuit/modules.dm]:
```suggestion
	materials = list(DEFAULT_WALL_MATERIAL = 30000, MATERIAL_PHORON = 12500, MATERIAL_SILVER = 10000, MATERIAL_GOLD = 10000)
```

File: code/modules/research/designs/mechfab/hardsuit/modules.dm [hunk 24]
```diff
@@ -8,125 +8,125 @@
...
+	materials = list(DEFAULT_WALL_MATERIAL = 7000, MATERIAL_GLASS = 5250)
 	build_path = /obj/item/rig_module/mounted/taser
 
 /datum/design/hardsuitmodules/egun_module
 	name = "mounted energy gun"
 	desc = "A forearm-mounted energy projector."
 	id = "egun_module"
 	req_tech = list(TECH_MATERIAL = 3, TECH_POWER = 4, TECH_COMBAT = 4, TECH_MAGNET = 3)
-	materials = list(DEFAULT_WALL_MATERIAL= 7000, "glass"= 2250, "uranium"= 3250, "gold"= 2500)
+	materials = list(DEFAULT_WALL_MATERIAL= 7000, MATERIAL_GLASS = 2250, MATERIAL_URANIUM = 3250, MATERIAL_GOLD = 2500)
```
Comment (Geevies) [inline code/modules/research/designs/mechfab/hardsuit/modules.dm]:
```suggestion
	materials = list(DEFAULT_WALL_MATERIAL = 7000, MATERIAL_GLASS = 2250, MATERIAL_URANIUM = 3250, MATERIAL_GOLD = 2500)
```

File: code/modules/research/designs/mechfab/hardsuit/modules.dm [hunk 25]
```diff
@@ -8,125 +8,125 @@
...
 	build_path = /obj/item/rig_module/mounted/egun
 
 /datum/design/hardsuitmodules/cooling_module
 	name = "mounted cooling unit"
 	desc = "A heat sink with liquid cooled radiator."
 	id = "cooling_module"
 	req_tech = list(TECH_MATERIAL = 2, TECH_POWER = 3, TECH_ENGINEERING = 3)
-	materials = list(DEFAULT_WALL_MATERIAL= 7000, "glass"= 5500)
+	materials = list(DEFAULT_WALL_MATERIAL= 7000, MATERIAL_GLASS = 5500)
```
Comment (Geevies) [inline code/modules/research/designs/mechfab/hardsuit/modules.dm]:
```suggestion
	materials = list(DEFAULT_WALL_MATERIAL = 7000, MATERIAL_GLASS = 5500)
```

File: code/game/objects/items/weapons/shields.dm [hunk 26]
```diff
@@ -62,7 +62,7 @@
 	throw_range = 4
 	w_class = 4.0
 	origin_tech = list(TECH_MATERIAL = 2)
-	matter = list("glass" = 7500, DEFAULT_WALL_MATERIAL = 1000)
+	matter = list(MATERIAL_GLASS = 7500, DEFAULT_WALL_MATERIAL = 1000)
```
Comment (Geevies) [inline code/game/objects/items/weapons/shields.dm]:
```suggestion
	matter = list(DEFAULT_WALL_MATERIAL = 1000, MATERIAL_GLASS = 7500)
```

File: code/game/objects/items/weapons/power_cells.dm [hunk 27]
```diff
@@ -26,7 +26,7 @@
 	throw_speed = 5
 	throw_range = 7
 	maxcharge = 1000
-	matter = list("metal" = 350, "glass" = 50)
+	matter = list("metal" = 350, MATERIAL_GLASS = 50)
```
Comment (Geevies) [inline code/game/objects/items/weapons/power_cells.dm]: "metal"

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 28]
```diff
@@ -579,7 +579,7 @@ var/global/list/obj/item/device/pda/PDAs = list()
 					"nitrogen" = "[round(n2_level*100,0.1)]",\
 					"oxygen" = "[round(o2_level*100,0.1)]",\
 					"carbon_dioxide" = "[round(co2_level*100,0.1)]",\
-					"phoron" = "[round(phoron_level*100,0.01)]",\
+					MATERIAL_PHORON = "[round(phoron_level*100,0.01)]",\
```
Comment (Geevies) [inline code/game/objects/items/devices/PDA/PDA.dm]: not a material

File: code/modules/mining/drilling/drill.dm (hunk 12 above)
Comment (NonQueueingMatt) [reply code/modules/mining/drilling/drill.dm]: This is fine.

**Pull request merged**